src/lib.rs

```diff
@@ -159,13 +159,18 @@ pub trait MiniscriptKey: Clone + Eq + Ord + fmt::Debug + fmt::Display + hash::Ha
     /// Used in the sha256 fragment
     type Sha256: Clone + Eq + Ord + fmt::Display + fmt::Debug + hash::Hash;
 
+    /// The associated [`hash256::Hash`] type for this [`MiniscriptKey`] type.
+    /// Used in the sha256 fragment
+    type Hash256: Clone + Eq + Ord + fmt::Display + fmt::Debug + hash::Hash;
+
```

Comment:
re: `Hash`, `Sha256`, and `Hash256`

From the docs on the trait its not really clear, at least to me, what is the relation between these three types. Two of them are named after fragments but one is not and they are all 'hashes' but there are other hash fragments as well. Can be add some documentation to try and help devs (*cough* me) to understand this?

Reply:
Yep, this is indeed clarification now that we have multiple hashes. I will add some clarification here.

Reply:
`Sha256` and `Hash256` are two of the five hash types supported by Script (the others are `Sha1` (not supported by miniscript), `Ripemd160` and `Hash160`). Meanwhile `Hash` is a trait from bitcoin_hashes which abstracts over all of these and allows feeding data into it.

(IIRC `Hash` is basically just a copy of `io::Write` but `io::Write` is std-only, for unclear reasons.)

Reply:
@apoelstra , I think the `Hash` is @tcharding is talking about here is MiniscirptKey::Hash which is different from `bitcoin_hashes::Hash`. 

The `Hash` represents the associated type corresponding to PkH fragment. Perhaps, with multiple hashes in the picture and considering #431  we should rename that to `RawPkHash`? 

I think we should re-evaluate whether we even need a Pk::Hash type anymore. That can be useful for abstractions over RawPkH in the future when we have partial descriptors. 

We can consider renaming that as a part of a separate PR.